### PR TITLE
Make infer datatype more robust

### PR DESCRIPTION
**Pull request by ly-kolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Review by dylangrandmont — APPROVED** [review]
Based on a possible scenario where multiple id_fields may be specified and within those combinations one of the fields may be empty (empty string or `None`), I think this change makes sense.

File: kolena/dataset/dataset.py [hunk 1]
```diff
@@ -99,7 +99,11 @@ def _normalize_url(x: str) -> str:
 
 
 def _infer_datatype_value(x: str) -> str:
```
Comment (yifanwu-kolena) [inline kolena/dataset/dataset.py]: i think we need to update the type hint to reflect the change

File: kolena/dataset/dataset.py [hunk 2]
```diff
@@ -99,7 +99,11 @@ def _normalize_url(x: str) -> str:
...
-    url = _normalize_url(x)
+    try:
+        url = _normalize_url(x or "")
+    except Exception:
+        return DatapointType.TABULAR.value
```
Comment (yifanwu-kolena) [inline kolena/dataset/dataset.py]: i think we can check the type of x, if not string just fallback to `DatapointType.TABULAR` to make the logic cleaner

File: tests/unit/dataset/test_dataset.py [hunk 3]
```diff
@@ -50,6 +50,8 @@
         ("test.pcd", DatapointType.POINT_CLOUD),
         ("gcp://summary.pdf", DatapointType.DOCUMENT),
         ("//my.mp3", DatapointType.AUDIO),
+        (None, DatapointType.TABULAR),
+        (123, DatapointType.TABULAR),
     ],
 )
 def test__infer_datatype_value(uri: str, expected: str) -> None:
```
Comment (yifanwu-kolena) [inline tests/unit/dataset/test_dataset.py]: type hint

File: kolena/dataset/dataset.py (hunk 2 above)
Comment (ly-kolena) [reply kolena/dataset/dataset.py]: Given that this is leveraging library, and we don't have all possible scenarios that could fail, I'm using try/exception over trying to if-condition all fail scenarios.

File: kolena/dataset/dataset.py (hunk 1 above)
Comment (ly-kolena) [reply kolena/dataset/dataset.py]: With the "implied" permissiveness of DataFrame input could be heterogeneous, in reality we're handling `Any`.

File: kolena/dataset/dataset.py (hunk 2 above)
Comment (yifanwu-kolena) [reply kolena/dataset/dataset.py]: if we use try-exception at here, assuming the type of x could be `Any`
- we will need to update `_normalize_url` to accept Any
- `urlparse` only accepts string, then we will do the type checking in `_normalize_url`

that's why i suggest to do the type checking at `_infer_datatype_value` level

Comment (ly-kolena) [reply kolena/dataset/dataset.py]: updated as suggested

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**